### DISCUSSION TITLE
ESP32 BLE + Xiaomi CGG1 - two questions

### DISCUSSION THREAD

**Discussion by MDo3** [arendst/Tasmota#15798] [Q&A] [answered]
Is it normal that my CGG1 sensor is permanently connected to the module instead of connecting to it at certain intervals, similar to LYWSD03MMC ? I have concerns about battery life.

![2022-06-15 07 14 13](https://user-images.githubusercontent.com/53032076/173750110-9eec12ac-5f59-430f-8fdd-62958d529384.jpg)

The second thing: If I enter the key in this form in the console, everything works, but only until the module is restarted:

`MI32Keys A4C138XXXXXX=E696D07178746441A9EXXXXXXEXXXXXX`

Is there a way to save the key permanently so that it works after rebooting the device?

**Comment by barbudor** [comment, marked as answer]
I'm not 100% sure but I think this is how CGG1 works vs   LYWS
The LYWS is much better especially with the Pvvx alternate firmware 

Most MI32 serttings are not persistent and the most convenient way of setting them on reboot is an `autoexec.bat` file with each command in an individual line
It is easy to create a file using the embedded file editor
This is how I do for my Mi32Option and BLEAlias
